controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -778,20 +824,48 @@ func (r *StorageClassClaimReconciler) reconcileCephFilesystemSubVolumeGroup() er
 	}
 
 	var cephFileSystemName string
+	var dataPoolName string
 	availableCephFileSystems := len(cephFilesystemList.Items)
 	if availableCephFileSystems == 0 {
 		return fmt.Errorf("no CephFileSystem found in the cluster")
 	}
-	if availableCephFileSystems > 1 {
-		klog.Warningf("More than one CephFileSystem found in the cluster, selecting the first one")
-	}
 
-	cephFileSystemName = cephFilesystemList.Items[0].Name
```

Comment:
Why did you decide to remove this chack?

Reply:
I moved this code to CreateOrUpdate call as here we are assuming will use the first filesystem in case if we have multiple filesystem we need to choose the right filesystem based on the profile